pkg/kube/client.go

```diff
@@ -44,16 +54,50 @@ const (
 	defaultLocalAddress = "localhost"
 )
 
-// Client is a helper for common Kubernetes client operations
-type Client interface {
+// BaseClient contains the core set of interfaces shared between all client users. This allows us to split
+// functions used only by CLI tools (like istioctl) and functions used only by servers (like Informers)
+type BaseClient interface {
```

Comment:
`BaseClient` would seem a little ugly to pass around. How about `Client` and `DynamicClient`?

Reply:
I was going to make this private but I didn't think that works well with embedding in public interfaces - maybe that is wrong. I don't anticipate this would be used by anyone

Reply:
how about a Client (what istiod uses) and an Extended client (what istioctl and tests use)?

Reply:
Yeah that works. I actually think the tests will likely only need the `Client`.